superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/GPUCore.cpp

```diff
@@ -0,0 +1,497 @@
+            get_rvalue_ptr(CD3DX12_RESOURCE_DESC::Buffer(persistentResourceSize)), D3D12_RESOURCE_STATE_COMMON, nullptr,
+            IID_PPV_ARGS(persistentBuffer.GetAddressOf())));
+
+        // The persistent resource should be bound as the output to the IDMLOperatorInitializer.
+        DML_BUFFER_BINDING bufferBinding{persistentBuffer.Get(), 0, persistentResourceSize};
+        DML_BINDING_DESC bindingDesc{DML_BINDING_TYPE_BUFFER, &bufferBinding};
+        m_bindingTable->BindOutputs(1, &bindingDesc);
+    }
+
+    ThrowIfFailed(m_dmlDevice->CreateCommandRecorder(IID_PPV_ARGS(&m_dmlCommandRecorder)));
+
+    // Record execution of the operator initializer.
+    m_dmlCommandRecorder->RecordDispatch(m_commandList.Get(), dmlOperatorInitializer.Get(), m_bindingTable.Get());
+    CloseExecuteResetWait();
+
+    // Bind and execute the operator on the GPU.
+    m_commandList->SetDescriptorHeaps(ARRAYSIZE(d3D12DescriptorHeaps), d3D12DescriptorHeaps);
+
+    // Reset the binding table to bind for the operator we want to execute (it was previously used to bind for the
+    // initializer).
+    dmlBindingTableDesc->Dispatchable = m_dmlCompiledOperator.Get();
+
+    ThrowIfFailed(m_bindingTable->Reset(dmlBindingTableDesc.get()));
+
+    if (temporaryResourceSize != 0) {
+        DML_BUFFER_BINDING bufferBinding{temporaryBuffer.Get(), 0, temporaryResourceSize};
+        DML_BINDING_DESC bindingDesc{DML_BINDING_TYPE_BUFFER, &bufferBinding};
+        m_bindingTable->BindTemporaryResource(&bindingDesc);
+    }
+
+    if (persistentResourceSize != 0) {
+        DML_BUFFER_BINDING bufferBinding{persistentBuffer.Get(), 0, persistentResourceSize};
+        DML_BINDING_DESC bindingDesc{DML_BINDING_TYPE_BUFFER, &bufferBinding};
+        m_bindingTable->BindPersistentResource(&bindingDesc);
+    }
+
+    CloseExecuteResetWait();
+}
+
+/**
+ * @brief Execute the computation GEMM op.
+ * @return the elapsed time in ms.
+ */
+template <typename T> double GPUCore::ExecuteComputeOp(int m, int n, int k) {
+
+    DML_BUFFER_BINDING inputBufferBindingA{m_inputBufferA.Get(), 0, sizeof(T) * m * k};
+    DML_BINDING_DESC inputBindingDescA{DML_BINDING_TYPE_BUFFER, &inputBufferBindingA};
+
+    DML_BUFFER_BINDING inputBufferBindingB{m_inputBufferB.Get(), 0, sizeof(T) * n * k};
+    DML_BINDING_DESC inputBindingDescB{DML_BINDING_TYPE_BUFFER, &inputBufferBindingB};
+
+    DML_BUFFER_BINDING bufferBinding = {nullptr, 0, 0};
+    DML_BINDING_DESC inputBindingDesc{DML_BINDING_TYPE_NONE, &bufferBinding};
+
+    std::array<DML_BINDING_DESC, 3> inputBindings = {inputBindingDescA, inputBindingDescB, inputBindingDesc};
+    m_bindingTable->BindInputs(3, inputBindings.data());
+
+    DML_BUFFER_BINDING outputBufferBinding{m_outputBuffer.Get(), 0, sizeof(T) * n * m};
+    DML_BINDING_DESC outputBindingDesc{DML_BINDING_TYPE_BUFFER, &outputBufferBinding};
+
+    m_bindingTable->BindOutputs(1, &outputBindingDesc);
```

Comment:
I think I have  concern on buffer management during the testing.

Reply:
moved this part into initializeOp